File: Data Structures & Algorithms/pow-x-n/submission-14.cpp
class Solution {
public:
    double myPow(double x, int n) {
        long long abs_n = n;
        if (abs_n < 0){
            abs_n = -abs_n;
        }
        double res = 1.0;
        while (abs_n > 0){
            if (abs_n & 1) {
                res = res * x;
            }
            x = x* x;
            abs_n >>= 1;
        }
        if (n < 0){
            return 1/res;
        }
        else{
            return res;
        }
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-14.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None: 
            return []
        res = []
        stack = deque([root])
        while len(stack) > 0:
            level = []
            level_size = len(stack)
            for _ in range(level_size):
                node = stack.popleft()
                level.append(node.val)
                if node.left is not None: 
                    stack.append(node.left)
                if node.right is not None: 
                    stack.append(node.right)
            res.append(level[-1])
        return res